compiler/rustc_middle/src/traits/specialization_graph.rs

```diff
@@ -211,7 +211,7 @@ pub fn ancestors<'tcx>(
     let specialization_graph = tcx.specialization_graph_of(trait_def_id);
 
     if specialization_graph.has_errored || tcx.type_of(start_from_impl).references_error() {
-        Err(ErrorReported)
+        Err(ErrorReported::new_i_know_what_im_doing())
```

Comment:
At the very least, `has_errored` should be an `Option<ErrorReported>`. But ideally, `references_error` could also give the field inside `ty::Error` (for that one we don't want to store it, so we might still need the unchecked constructor).